api/handlers.go

```diff
@@ -139,8 +139,10 @@ func (si *ServerImplementation) MakeHealthCheck(ctx echo.Context) error {
 		errors = append(errors, fmt.Sprintf("database error: %s", health.Error))
 	}
 
-	if si.fetcher != nil && si.fetcher.Error() != "" {
-		errors = append(errors, fmt.Sprintf("fetcher error: %s", si.fetcher.Error()))
+	if si.dataError != nil {
+		if err := si.dataError(); err != nil {
```

Comment:
```suggestion
		if si.dataError != nil && si.dataError() != nil {
		}
```

Reply:
The reason I've separated it out is that we need to add `err` to the set of known errors. We could just call `si.dataError()` again inside the block to reference the error, but because the pipeline will keep retrying it's possible that the first call would show an error but the second call might return `nil`.

This way we have a single call which either returns an error that is added to the health check or it's `nil`.